lib/plsql/jdbc_connection.rb

```diff
@@ -348,7 +348,7 @@ def plsql_to_ruby_data_type(metadata)
         [Java::OracleSql::CLOB, nil]
       when "BLOB"
         [Java::OracleSql::BLOB, nil]
-      when "NUMBER"
+      when "NUMBER", "BINARY_FLOAT", "BINARY_DOUBLE"
```

Comment:
Some wondering whether these should be `BigDecimal` or `Float` (on native level Float seems closer equivalent). There are some intricacies when comparing floating-point numbers though - as they're approximate.